compiler-core/src/language_server/completer.rs

```diff
@@ -916,3 +967,120 @@ fn field_completion(label: &str, type_: Arc<Type>) -> CompletionItem {
         ..Default::default()
     }
 }
+
+pub struct LocalCompletion<'a> {
+    mod_name: &'a str,
+    insert_range: Range,
+    cursor: u32,
+    completions: Vec<CompletionItem>,
+}
+
+impl<'a> LocalCompletion<'a> {
+    pub fn new(mod_name: &'a str, insert_range: Range, cursor: u32) -> Self {
+        Self {
+            mod_name,
+            insert_range,
+            cursor,
+            completions: Vec::new(),
+        }
+    }
+
+    pub fn completions(&self) -> Vec<CompletionItem> {
+        self.completions.clone()
+    }
+
+    pub fn visit_statements(&mut self, statements: &'a [Statement<Arc<Type>, TypedExpr>]) {
```

Comment:
Could you use detailed comments to document why this function does this please 🙏  I didn't understand it at first.

Reply:
I refactored a little bit to make it clearer. Now it receives the Function instead of the body statements so you don't have to check the usage to know what it does! This also means that suggestions for the function argument are added here. Of course, with all these changes I changed the name to clarify and added some comments to document it. Let me know what you think!